core/src/test/java/jenkins/security/HMACConfidentialKeyTest.java

```diff
@@ -38,4 +40,39 @@ public void loadingExistingKey() {
         }
     }
 
+
+    @Test
+    public void testMacWithShortenedCodeOnFips() {
+        System.setProperty("jenkins.security.FIPS140.COMPLIANCE", "true");
+        HMACConfidentialKey key1 = new HMACConfidentialKey("test", 16);
+        try {
+            String str = key1.mac("Hello World");
+        } catch (IllegalArgumentException exception) {
+            assertEquals("Supplied length can't be less than 32 on FIPS mode", exception.getMessage());
+        }
+    }
+
+    @Test
+    public void testMacWithShortenedCodeOnNonFips() {
```

Comment:
naming is hard - but the lenght/size is shortened not the code?
`testTruncatedMacOnNonFips` ?